### ISSUE TITLE
Use smaller test collections

### ISSUE THREAD

**Issue by dougbu** [aspnet/AspNetWebStack#95]
The fix for #41 was intended to be temporary. It worked but had the cost of placing a large number of tests in the same collection, preventing parallel execution of the affected test classes.

Should revert f0988360 and instead use `[CollectionDefinition(DisableParallelization = true)]` for either the original collection or a new collection containing `DisplayExtensionsTest`, `EditorExtensionsTest` and any other test class that writes to the `ScopeStorage`, `ViewEngines.Engines` or `ModelMetadataProviders.Current` `static` properties.

FYI:
- `DisableParallelization` is a new feature of the xUnit 2.3.0 runtime. It ensures no other test collection (aka test class in most cases) runs in parallel with one using that `CollectionDefinition`.
- f0988360 affected only the System.Web.Mvc.Test test assembly but should check the other assemblies to see if  `DisableParallelization` would help there too i.e. see if only a small subset of the test collections change `static` properties
- no test assembly has more than one explicit test collection i.e. almost every test class runs in parallel with others
- multiple test assemblies declare a `"Uses ScopeStorage or ViewEngines.Engines"` test collection but, due to the assembly boundaries, they aren't really the same

**Comment by dougbu** [comment]
Moving to 3.2.5 and self-assigning since I'll do this as my first try on fixing #125. That is, #125 may be a duplicate of this issue.
